core/src/main/java/org/apache/carbondata/core/index/TableIndex.java

```diff
@@ -207,15 +208,22 @@ public CarbonTable getTable() {
       Map<Segment, List<Index>> indexes) throws IOException {
     Set<String> missingSISegments = filter.getMissingSISegments();
     for (Segment segment : segments) {
+      List<Index> segmentIndices = indexes.get(segment);
       if (segment == null ||
-          indexes.get(segment) == null || indexes.get(segment).isEmpty()) {
+          indexes.get(segment) == null || segmentIndices.isEmpty()) {
         continue;
       }
       boolean isExternalOrMissingSISegment = segment.isExternalSegment() ||
           (missingSISegments != null && missingSISegments.contains(segment.getSegmentNo()));
       List<Blocklet> pruneBlocklets = new ArrayList<>();
-      SegmentProperties segmentProperties =
-          segmentPropertiesFetcher.getSegmentProperties(segment, partitionLocations);
+      SegmentProperties segmentProperties;
+      if (segmentIndices.get(0) instanceof BlockIndex) {
```

Comment:
Need to loop here and check if any of the segmentIndices is BlockIndex instead of always checking the first index?

Reply:
Also CC: @Indhumathi27 , you can also review once. 
May be we can remove   `segmentPropertiesFetcher.getSegmentProperties(segment, partitionLocations); `
and always call `segmentPropertiesFetcher.getSegmentPropertiesFromIndex` as we have indexes already at this place.

Reply:
@ajantha-bhat  @Indhumathi27 1) we are getting segment properties from one blockindex which is same for all. 
2) In this flow we may receive SI and bloom indexes also which will prepare by SI , bloom index factory [List(SecondaryIndex) or List(BloomIndex)]. In this case we should use "segmentPropertiesFetcher.getSegmentProperties" as they are not instance of BlockIndex. inside getSegmentProperties method we are getting and preparing list of BlockIndex and getting segment propteries from block index.

I will add scenario description in code and push.

Reply:
I mean line 211, when `indexes.get(segment)` is done, all indexes will be present in this list. Like blockindex, SI, bloom, lucence. 
so, we just have to find blockindex (which is always be present even for SI table) and get the the segment properties ? 
@Indhumathi27 : Is my understanding correct or we need both cases here ?

Reply:
i checked in SI flow, it has only SI index info and does not have blockindex info. We can add separate checks

Reply:
When cgIndexExprWrapper.prune is called (for SI/Bloom), in that case, it will have either BloomcoarseGainIndex or SIindex. So, we have to keep both cases